### PR TITLE
Lazily populate the Commissioner's callback cache.

### PR DESCRIPTION
**Pull request by nelhage**
This is worth a ~20% speedup on a large private code base I work on,
presumably because many Commissioners never actually encounter all
possible node types.

### CODE REVIEW

File: lib/rubocop/cop/commissioner.rb [hunk 1]
```diff
@@ -69,9 +72,6 @@ def remove_irrelevant_cops(filename)
 
       def setup_callbacks
```
Comment (bbatsov) [inline lib/rubocop/cop/commissioner.rb]: The name of this name becomes kind of confusing now.

**Comment by bbatsov** [comment]
Nice catch. Apart from my inline remark - please, drop the `.` from commit message's title.

**Comment by alexdowad** [comment]
I had also noticed that this was a hotspot, but didn't think of cooling it down this way. Thanks.

**Comment by nelhage** [comment]
(bumped; not sure if pushing a commit notified subscribers, and sorry for the redundant email if so)

**Pull request merged**